### PR TITLE
chore: add warning about database availability

### PR DESCRIPTION
**Pull request by johnstcn**
This commit adds a warning regarding Coder's hard dependency on its PostgreSQL database.
EDIT: `prettier` also apparently decided 2 spaces now instead of 4. 🤷

### CODE REVIEW

File: setup/installation.md [hunk 1]
```diff
@@ -31,7 +36,8 @@ resources from
 
 ## Install Coder
 
-1. Create the Coder [namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/):
+1. Create the Coder
+   [namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/):
```
Comment (johnstcn) [inline setup/installation.md:40]: prettier

File: setup/installation.md [hunk 2]
```diff
@@ -90,142 +96,143 @@ resources from
...
+  passwordSecret:
+    # coderd.superAdmin.passwordSecret.name -- Name of a secret that should
+    # be used to determine the password for the super admin account. The
+    # password should be contained in the field `password`, or the manually
+    # specified one.
+    name: ""
+    # coderd.superAdmin.passwordSecret.key -- The key of the secret that
+    # contains the super admin password.
+    key: "password"
+```
```
Comment (johnstcn) [inline setup/installation.md:114]: prettier

File: setup/installation.md [hunk 3]
```diff
@@ -90,142 +96,143 @@ resources from
...
+> users on your system through `/proc`. If this is a concern, you can use
+> `--from-file=password=/dev/stdin` instead of `--from-literal=...` to enter
+> your password and press `Ctrl+D` when you're done to submit it.
+>
+> Ensure that there are no trailing white spaces in your password secret.
+
+For more detailed configuration instructions,
+[see our PostgreSQL setup guide](../guides/deployments/postgres.md).
+Alternatively, see our
+[guide on connecting to AWS RDS via IAM credentials](/guides/admin/awsrds.md).
```
Comment (johnstcn) [inline setup/installation.md:154]: prettier

File: setup/installation.md [hunk 4]
```diff
@@ -90,142 +96,143 @@ resources from
...
 
 ## Enable dev URLs
 
 **Optional**: Enable dev URL usage.
-   [You must provide a wildcard domain in the Helm chart](../admin/devurls.md).
+[You must provide a wildcard domain in the Helm chart](../admin/devurls.md).
```
Comment (johnstcn) [inline setup/installation.md:159]: prettier

File: setup/installation.md [hunk 5]
```diff
@@ -90,142 +96,143 @@ resources from
...
 
-   ```yaml
-   coderd:
-     devurlsHost: "*.my-custom-domain.io"
-   ```
+```yaml
+coderd:
+  devurlsHost: "*.my-custom-domain.io"
+```
```
Comment (johnstcn) [inline setup/installation.md:164]: prettier

File: setup/installation.md [hunk 6]
```diff
@@ -90,142 +96,143 @@ resources from
...
 
 ## Enable TLS
 
 **Optional:** To set up TLS:
 
-   a. You will need to create a TLS secret. To do so, run the following with the
-   `.pem` files provided by your certificate:
+a. You will need to create a TLS secret. To do so, run the following with the
+`.pem` files provided by your certificate:
```
Comment (johnstcn) [inline setup/installation.md:171]: prettier

File: setup/installation.md [hunk 7]
```diff
@@ -90,142 +96,143 @@ resources from
...
 
-   ```console
-   kubectl create secret tls tls-secret --key key.pem --cert cert.pem
-   ```
+```console
+kubectl create secret tls tls-secret --key key.pem --cert cert.pem
+```
```
Comment (johnstcn) [inline setup/installation.md:175]: prettier

File: setup/installation.md [hunk 8]
```diff
@@ -90,142 +96,143 @@ resources from
...
 
-   > If your certificate provider does not provide `.pem` files, then you may
-   > need to attach the certificate to the LoadBalancer manually.
+> If your certificate provider does not provide `.pem` files, then you may need
+> to attach the certificate to the LoadBalancer manually.
 
-   b. Attach the secret to the `coderd` service by setting the following values:
+b. Attach the secret to the `coderd` service by setting the following values:
```
Comment (johnstcn) [inline setup/installation.md:180]: prettier

File: setup/installation.md [hunk 9]
```diff
@@ -90,142 +96,143 @@ resources from
...
-     tls:
-       hostSecretName: <tls-secret>
-       devurlsHostSecretName: <tls-secret>
-   ```
+```yaml
+coderd:
+  tls:
+    hostSecretName: <tls-secret>
+    devurlsHostSecretName: <tls-secret>
+```
```
Comment (johnstcn) [inline setup/installation.md:187]: prettier

File: setup/installation.md [hunk 10]
```diff
@@ -90,142 +96,143 @@ resources from
...
+    devurlsHostSecretName: <tls-secret>
+ingress:
+  enable: true
+  # Hostname to use for routing decisions
+  host: "coder.coderhost.com"
+  # Custom annotations to apply to the resulting Ingress object
+  # This is useful for configuring other controllers in the cluster
+  # such as cert-manager or the ingress controller
+  annotations: {}
+```
```
Comment (johnstcn) [inline setup/installation.md:215]: prettier

File: setup/installation.md [hunk 11]
```diff
@@ -90,142 +96,143 @@ resources from
...
 
 ## Configure a proxy
 
-**Optional:** To have Coder initiate outbound connections via a proxy, set
-   the following (applicable) values:
+**Optional:** To have Coder initiate outbound connections via a proxy, set the
+following (applicable) values:
```
Comment (johnstcn) [inline setup/installation.md:220]: prettier

File: setup/installation.md [hunk 12]
```diff
@@ -90,142 +96,143 @@ resources from
...
-         https: ""
-         exempt: "cluster.local"
-   ```
+```yaml
+coderd:
+  proxy:
+    http: ""
+    https: ""
+    exempt: "cluster.local"
+```
```
Comment (johnstcn) [inline setup/installation.md:228]: prettier

**Review by ericpaulsen — APPROVED** [review]

**Review by sharkymark — APPROVED** [review]

**Pull request merged**